ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java

```diff
@@ -70,78 +72,81 @@ class AdminShapeRestControllerTest {
 	@Autowired
 	private MockMvc mockMvc;
 
+	@Autowired
+	private ShaclCollection shaclCollection;
+
 	@BeforeEach
     void setUp() {
         when(ldesConfigShaclValidator.supports(any())).thenReturn(true);
     }
 
-	@Test
-	void when_StreamAndShapeArePresent_Then_ShapeIsReturned() throws Exception {
-		String collectionName = "name1";
-		Model expectedShapeModel = readModelFromFile("shape-1.ttl");
-		LdesConfigModel shapeConfigModel = LdesConfigModel.createLdesConfigShape(collectionName, expectedShapeModel);
-		when(ldesConfigModelService.retrieveShape(collectionName)).thenReturn(shapeConfigModel);
-		ResultActions resultActions = mockMvc.perform(get("/admin/api/v1/eventstreams/" + collectionName + "/shape"))
-				.andDo(print())
-				.andExpect(status().isOk());
-		MvcResult result = resultActions.andReturn();
-		Model actualModel = RdfModelConverter.fromString(result.getResponse().getContentAsString(), Lang.TURTLE);
-		Assertions.assertTrue(actualModel.isIsomorphicWith(expectedShapeModel));
-	}
+	@Nested
+	class GetRequest {
+		@Test
+		void when_ShapeIsPresentArePresent_Then_ShapeIsReturned() throws Exception {
+			String collectionName = "name1";
+			Model expectedShapeModel = readModelFromFile("shape-1.ttl");
+			when(shaclCollection.retrieveShape(collectionName))
+					.thenReturn(new LdesConfigModel(collectionName, expectedShapeModel));
+
+			ResultActions resultActions = mockMvc
+					.perform(get("/admin/api/v1/eventstreams/" + collectionName + "/shape"))
+					.andDo(print())
+					.andExpect(status().isOk());
+
+			MvcResult result = resultActions.andReturn();
```

Comment:
I have seen the below lines on a couple of places, perhaps we should consider using our own matcher. Then we could just add `.andExpect(content().string(IsIsomorphic.with(model)));` to the chain above and be done.

Reply:
Possible implementation for the matcher:

```
public class IsIsomorphic extends BaseMatcher<String> {

    private final Model model;

    IsIsomorphic(Model model) {
        this.model = model;
    }

    public static IsIsomorphic with(Model model) {
        return new IsIsomorphic(model);
    }

    @Override
    public boolean matches(Object object) {
        if (object instanceof String string) {
            Model actualModel = RdfModelConverter.fromString(string, Lang.TURTLE);
            return actualModel.isIsomorphicWith(model);
        }
        return false;
    }

    @Override
    public void describeTo(Description description) {
        description.appendText("Result should be isomorphic with provided model");
    }
}
```

Reply:
Ha, nice one, well actually in the follow-up PR we fix it by including the `isIsomorphicWith` in the `equals` method of the object :-)